### PR TITLE
Added enable functionality

### PR DESCRIPTION
**Pull request by feschiessl**
* Added constructor functions with optional enable pin
* Added class functions to enable/disable motor
* Added lines to Arduino examples
* enable / sleep / reset controls

### CODE REVIEW

File: src/BasicStepperDriver.cpp [hunk 1]
```diff
@@ -102,3 +108,18 @@ int BasicStepperDriver::rotate(double deg){
     int steps = deg * motor_steps * microsteps / 360;
     return move(steps);
 }
+
+/*
+ * Enable/Disable the motor by setting a digital flag
+ */
+void BasicStepperDriver::enable(void){
+    pinMode(enable_pin, OUTPUT);
```
Comment (laurb9) [inline src/BasicStepperDriver.cpp]: Setting up pin mode should be done in init, so it's not floating. The only question is should it be enabled or disabled at start.

File: src/BasicStepperDriver.cpp [hunk 2]
```diff
@@ -102,3 +108,18 @@ int BasicStepperDriver::rotate(double deg){
...
+    digitalWrite(enable_pin, LOW);
```
Comment (laurb9) [inline src/BasicStepperDriver.cpp]: if IS_CONNECTED(enable_pin):...

File: src/BasicStepperDriver.h [hunk 3]
```diff
@@ -37,6 +37,8 @@ class BasicStepperDriver {
     int rpm;
     int dir_pin;
     int step_pin;
+    int enable_pin;
```
Comment (laurb9) [inline src/BasicStepperDriver.h]: int enable_pin = PIN_UNCONNECTED;

**Comment by laurb9** [comment]
Thanks. The only problem is that if we want to add sleep functionality later, it will take another parameter that will now be ambigous (or we'll have to specify both enable and sleep).
Also I personally would leave sleep out.

**Comment by feschiessl** [comment]
I committed changes with the default motor being enabled. I think that's the most natural thing, no?! And with the additional sleep option, I would say that one option for the same outcome is enough...

**Pull request merged**